integration/service/create_test.go

```diff
@@ -309,6 +312,144 @@ func TestCreateServiceConfigFileMode(t *testing.T) {
 	assert.NilError(t, err)
 }
 
+// TestServiceCreateSysctls tests that a service created with sysctl options in
+// the ContainerSpec correctly applies those options.
+//
+// To test this, we're going to create a service with the sysctl option
+//
+//   {"net.ipv4.ip_nonlocal_bind": "0"}
+//
+// which is going to run a command to cat the value of that parameter into the
+// service logs, and we'll verify that it is 0 as expected.
+//
+// Next, we'll remove that service and create a new service with that option
+// set to 1. This means that no matter what the default is, we can be confident
+// that the sysctl option is applying as intended.
+//
+// Additionally, we'll do a service inspect to verify that the inspect output
+// includes the desired sysctl option.
+//
+// We're using net.ipv4.ip_nonlocal_bind because it's something that I'm fairly
+// confident won't be modified by the container runtime, and won't blow
+// anything up in the test environment
+func TestCreateServiceSysctls(t *testing.T) {
+	defer setupTest(t)()
+	d := swarm.NewSwarm(t, testEnv)
+	defer d.Stop(t)
+	client := d.NewClientT(t)
+	defer client.Close()
+
+	ctx := context.Background()
+
+	// run thie block twice, so that no matter what the default value of
+	// net.ipv4.ip_nonlocal_bind is, we can verify that setting the sysctl
+	// options works
+	for _, expected := range []string{"0", "1"} {
+
+		// store the map we're going to be using everywhere.
+		sysctlOpts := map[string]string{"net.ipv4.ip_nonlocal_bind": expected}
+
+		// Create the service with the sysctl options
+		var instances uint64 = 1
+		serviceName := "TestService_" + t.Name() + "_" + expected
+		serviceID := swarm.CreateService(t, d,
+			swarm.ServiceWithName(serviceName),
+			swarm.ServiceWithCommand([]string{
+				"/bin/sh", "-c",
+				"cat /proc/sys/net/ipv4/ip_nonlocal_bind && sleep 10",
+			}),
+			swarm.ServiceWithReplicas(instances),
+			swarm.ServiceWithSysctls(sysctlOpts),
+		)
+
+		// wait for the service to converge to 1 running task as expected
+		poll.WaitOn(t, serviceRunningTasksCount(client, serviceID, instances))
+
+		// get all of the tasks of the service, so we can get the container
+		// logs
+		filter := filters.NewArgs()
+		filter.Add("service", serviceID)
+		tasks, err := client.TaskList(ctx, types.TaskListOptions{
+			Filters: filter,
+		})
+		assert.NilError(t, err)
+		assert.Check(t, is.Equal(len(tasks), 1))
+
+		// just because the container started, doesn't mean it's actually run
+		// the command yet. if we try to read the logs instantly, we might be
+		// too fast and read before the output of the cat is available. to
+		// avoid that case, we're using poll.WaitOn with a closure to wait
+		// until logs actually respond
+		poll.WaitOn(t, func(t poll.LogT) poll.Result {
+			body, err := client.ContainerLogs(
```

Comment:
Should we use service logs instead?

Reply:
As the person ostensibly responsible for service logs, no.